## 문제
- 주어진 데이터(data/toothgrowth.csv)에는 기니피그 60마리의 치아 길이, 투여한 비타민 C 종류, 투여량이 저장되어 있다.  오렌지주스를 투여받은 기니피그의 치아 길이 평균이 아스코르브산을 투여받은 기니피그의 치아 길이 평균과 일치하는지 t-검정(Two Sample t-test)를 통해 답하고자 한다. 가설은 아래와 같다.
- 가설검정
    + 귀무가설 : 오렌지주스 투여 기니피그 치아 길이 평균과 아스코르브산 투여 받은 기니피그의 치아길이 평균의 차이는 0이다.
    + 대립가설 : 오렌지주스 투여 기니피그 치아 길이 평균과 아스코르브산 투여 받은 기니피그의 치아길이 평균의 차이는 0이 아니다. (둘 평균 다르다.)
- 데이터의 변수
    + len : 치아의 길이
    + supp : 투여한 약제(VC: 아스코르브산, OJ: 오렌지주스)
    + does : 투여량
- 단, 데이터의 각 변수들은 정규분포를 만족하며, 두 그룹은 등분산을 띤다.
- 문제 1: 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.
- 문제 2: 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)
- 문제 3: 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.
      

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import scipy.stats as stats
import math
from statistics import NormalDist
from scipy.stats import ttest_1samp
from scipy import stats

## 데이터 불러오기

In [3]:
import pandas as pd
data = pd.read_csv("c:/Archiving/IMDGB/step03_statistics/dataset/toothgrowth.csv")
data.head()

len supp  dose
0   4.2   VC   0.5
1  11.5   VC   0.5
2   7.3   VC   0.5
3   5.8   VC   0.5
4   6.4   VC   0.5

## 가설검정
- 독립표본 T-검정은 정규성 검정 뿐만 아니라, 두 그룹이 등분산성을 띠는지 확인하는 작업 필요
- Levene 검정을 사용한 등분산성 검정이 선행되어야 함
- 가설설정
  + 귀무가설: 두 그룹의 분산 같다
  + 대립가설: 두 그룹의 분산은 같지 않다.

In [8]:
data.loc[data['supp']=='OJ']

len supp  dose
30  15.2   OJ   0.5
31  21.5   OJ   0.5
32  17.6   OJ   0.5
33   9.7   OJ   0.5
34  14.5   OJ   0.5
35  10.0   OJ   0.5
36   8.2   OJ   0.5
37   9.4   OJ   0.5
38  16.5   OJ   0.5
39   9.7   OJ   0.5
40  19.7   OJ   1.0
41  23.3   OJ   1.0
42  23.6   OJ   1.0
43  26.4   OJ   1.0
44  20.0   OJ   1.0
45  25.2   OJ   1.0
46  25.8   OJ   1.0
47  21.2   OJ   1.0
48  14.5   OJ   1.0
49  27.3   OJ   1.0
50  25.5   OJ   2.0
51  26.4   OJ   2.0
52  22.4   OJ   2.0
53  24.5   OJ   2.0
54  24.8   OJ   2.0
55  30.9   OJ   2.0
56  26.4   OJ   2.0
57  27.3   OJ   2.0
58  29.4   OJ   2.0
59  23.0   OJ   2.0

In [10]:
data.loc[data['supp']=='OJ','len']

30    15.2
31    21.5
32    17.6
33     9.7
34    14.5
35    10.0
36     8.2
37     9.4
38    16.5
39     9.7
40    19.7
41    23.3
42    23.6
43    26.4
44    20.0
45    25.2
46    25.8
47    21.2
48    14.5
49    27.3
50    25.5
51    26.4
52    22.4
53    24.5
54    24.8
55    30.9
56    26.4
57    27.3
58    29.4
59    23.0
Name: len, dtype: float64

In [14]:
# levene 검정 for 등분산성 검정
from scipy import stats
#stats.levene(data.loc[data['supp']=='OJ',len] ,data.loc[data['supp']=='VC', 'len'])

In [30]:
# 다시 해보자
from scipy import stats
stats.levene( data.loc[data['supp']=='OJ', 'len'] , data.loc[data['supp']=='VC', 'len'] )

LeveneResult(statistic=np.float64(1.2135720656945064), pvalue=np.float64(0.2751764616144052))

- p_value가 0.275 이므로 귀무가설을 채택한다. -> 그래서 t-test 할 수 있게 된거임

## 문제 1
- 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.

In [16]:
oj_mean = data.loc[data['supp']=='OJ', 'len'].mean()
round(oj_mean,2)

np.float64(20.66)

In [18]:
vc_mean = data.loc[data['supp']=='VC', 'len'].mean()
round(vc_mean,2)

np.float64(16.96)

In [20]:
# 두 집단 평균 차이가 남

## 문제 2
- 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)

In [23]:
from scipy import stats
t, p = stats.ttest_ind(data.loc[data['supp']=='OJ', 'len'], 
                       data.loc[data['supp']=='VC', 'len'], 
                       equal_var=True) # equal_var = True, when 등분산성 만족할 때
t, p

(np.float64(1.91526826869527), np.float64(0.06039337122412848))

In [24]:
# 왜 결과 다르징 . 여튼

In [28]:
# p value가 0.06 -> 귀무가설 채택... = 뭘 투여 하느냐에 따라 두 집단 평균 길이 차이 없다. 아니 근데 0.05보다 큰데 왜 채택하는거임? .......? ...
# 아 커야 채택하는 구나.


In [ ]:
#

-1.9153 0.06


## 문제 3.
- 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.

In [26]:
round(p,4)

np.float64(0.0604)

In [ ]:
#

0.0604


In [ ]:
#

채택


In [29]:
if p>= 0.05:
  print("귀무가설 채택")
else:
  print("귀무가설 기각 ==> 대립가설 채택")

귀무가설 채택
